test_files/async_functions/async_functions.ts

```diff
@@ -52,11 +52,10 @@ class Container {
 }
 
 // TODO(#1099): uncomment this test after the async emit is fixed.
-// const asyncFnExpression = async function f() {};
+// export const asyncFnExpression = async function f() {};
 
-const asyncArrowFn = async () => {};
+export const asyncArrowFn = async () => {};
```

Comment:
I don't understand the test change here. Why does this need to be exported, and why do we drop the `this` type from the function below? IIRC that was intentional to test the behaviour when a `this` type is set (`hasExistingTag`).

Reply:
... but maybe should not use `this: unknown` but something like `this: SomeType`.

Reply:
I added 'export' everywhere because that triggers yet more JSCompiler subtlety, but I see it is confusing, so I fixed it by duplicating these.

It was not intentional that the lower function had a 'this' type -- that's why there was the TODO.  We had been adding "this: unknown" in g3 to make functions like this one compile, and this test was verifying that fix.  But this patch now fixes top-level async functions, so I can now remove the 'this' bit.

I added an additional function with a 'this' tag for that test coverage.